proposals/NNNN-GetInputScreen.md

```diff
@@ -0,0 +1,140 @@
+  </param>
+    <param name="softButtons" type="Common.SoftButton" minsize="0" maxsize="8" array="true" mandatory="false">
+    <description>
+      App defined SoftButtons for Cancel, edit, submit, etc.
+    </description>
+  </param>
+</function>
+```
+
+Add a new struct called `Field` and an enum called `FieldType` to get user input in the GetInput screen.
+This enables text input fields, numeric input fields and combo box input fields.  Text and numberic input fields would bring up a keyboard or numpad. Combo box input would bring up a drop down list of items.  
+```xml
+<struct name="Field">
+  <description>
+    A single field for data input
+  </description>
+  <param name="type" type="Common.FieldType" mandatory="true">
+  </param>
+  <param name="label" maxlength="500" type="String" mandatory="true">
+  </param>
+  <param name="image" type="Common.Image" mandatory="false">
+    <description>
+      An image icon for the input field
+    </description>
+  </param>
+  <param name="text" maxlength="500" type="String" mandatory="false">
+    <description>
+      The text that is already present in the input field
+    </description>
+  </param>
+  <param name="format" maxlength="500" type="String" mandatory="false">
+    <description>
+      A text string used to format the input text to a certain style. Any characters except underscore will remain in the field with no way to type over them. The underscore character will represent a space that the user can type in.  E.g. A format string for Phone number would be +_(___)___-____
+    </description>
+  </param>
+  <param name="additionalInfo" maxlength="500" type="String" mandatory="false">
+    <description>
+      More info that is shown by the input field
+    </description>
+  </param>
+  <param name="error" type="Boolean" mandatory="false">
+    <description>
+      If set to true, then the field has some problem with it and the HMI should highlight it in red.
+    </description>
+  </param>
+  <param name="comboBoxChoices" type="String" maxlength="500" array="true" minsize="0" maxsize="200" mandatory="false">
+    <description>
+      List of choices if the type is COMBO_BOX. Otherwise ignored.
+    </description>
+  </param>
+  <param name="inputMask" type="KeyboardInputMask" mandatory="false">
+    <description>
+      Used to cover up personal information like credit card number. Defaults to "DISABLE_INPUT_KEY_MASK"
+    </description>
+  </param>
+</struct>
+```
+```xml
+  <enum name="FieldType" since="X.X">
+    <description>The type of input field</description>
+    <element name="COMBO_BOX" />
+    <element name="TEXT" />
+    <element name="NUMBER" />
+      <description>
+        This is a notice for the HMI to bring up a keyboard, however the response for this field will still be a string.
+      </description>
+  </enum>
+```
+
+And the response which would be sent when the user presses a softbutton - this would use RPC encryption which is handled by policies.
```

Comment:
```suggestion
Regarding the response which would be sent when the user presses a softbutton, this would use RPC encryption which is handled by policies.
```